site/pages/api/add-user.api.ts

```diff
@@ -0,0 +1,67 @@
+import { UsernameExistsException } from "@aws-sdk/client-cognito-identity-provider";
+import { NextApiRequest, NextApiResponse } from "next";
+import { ADD_USER_PAGE_PATH, COOKIES_ADD_USER_ERRORS, USER_MANAGEMENT_PAGE_PATH } from "../../constants";
+import { createUser } from "../../data/cognito";
+import { addUserSchema } from "../../schemas/add-user.schema";
+import { flattenZodErrors } from "../../utils";
+import {
+    redirectToError,
+    setCookieOnResponseObject,
+    redirectTo,
+    destroyCookieOnResponseObject,
+} from "../../utils/apiUtils";
+
+const addUser = async (req: NextApiRequest, res: NextApiResponse) => {
```

Comment:
I've mentioned this in Alaina's PR but I'll put it here as well: We need to restrict access to this api endpoint so only authenticated admins can call it, if we update line 173 in middleware.api.ts to:
```ts
if (request.nextUrl.pathname.startsWith("/admin/") || request.nextUrl.pathname.startsWith("/api/admin/")) {
```
and move this file into an admin folder then that should do the job, the action endpoint for the form will need updating too

Reply:
done